mkosi/__init__.py

```diff
@@ -2508,33 +2540,43 @@ def install_alma_repos(args: CommandLineArguments, root: Path, epel_release: int
     setup_dnf(args, root, repos)
 
 
+def parse_epel_release(release: str) -> Tuple[int, bool]:
+    fields = release.split(".")
+    if fields[0].endswith("-stream"):
+        is_stream = True
+        epel_release = fields[0].split("-")[0]
+    else:
+        is_stream = False
+        epel_release = fields[0]
+
+    return int(epel_release), is_stream
+
+
 def is_older_than_centos8(release: str) -> bool:
     # CentOS 7 contains some very old versions of certain libraries
     # which require workarounds in different places.
     # Additionally the repositories have been changed between 7 and 8
-    epel_release = release.split(".")[0]
-    try:
-        return int(epel_release) <= 7
-    except ValueError:
-        return False
+    epel_release, is_stream = parse_epel_release(release)
+    return epel_release <= 7
 
 
 @complete_step("Installing CentOS…")
 def install_centos(args: CommandLineArguments, root: Path, do_run_build_script: bool) -> None:
-    old = is_older_than_centos8(args.release)
-    epel_release = int(args.release.split(".")[0])
+    epel_release, is_stream = parse_epel_release(args.release)
 
-    if old:
+    if epel_release <= 7:
         install_centos_repos_old(args, root, epel_release)
-    else:
+    elif epel_release <= 8:
         install_centos_repos_new(args, root, epel_release)
+    else:
+        install_centos_stream_repos(args, root, epel_release)
```

Comment:
Two things here:
1. `is_older_than_centos8` is no longer used. Either use it or remove it
2. I'm not sure I fully understand the logic here, but that might be my fault, because there's now so many CentOS versions in flight, I can't keep up anymore. :) Isn't there now CentOS 8 Stream and CentOS 9 Stream? I'm not quite sure I understand how this meshes with EPEL, but wouldn't I end up in the middle branch when trying to install CentOS 8 Stream? I don't think I see `is_stream` being used.

Reply:
`is_older_than_centos8` is used in a bunch of other functions, that's why I left it in place for now; happy to codemod it out though if you prefer

`epel_release` is misnamed; while it may have originally been used to denote the EPEL version, it looks like its actual meaning (even before my PR) was to be the integer version of the RHEL-like distro that's being processed. It probably should be renamed (maybe to `el_release`?), but I didn't want to do it here as it would have made the PR harder to review; again, happy to codemod it if you prefer

About what's going on with CentOS, here's the short version. There's two distros, CentOS Linux and CentOS Stream. CentOS Linux is what used to be called just CentOS, and is a straight rebuilt of RHEL; its final release (CentOS Linux 8) goes EOL on Dec 31; it is served from mirror.centos.org via mirrorlist. CentOS Stream is a new distribution that tracks the next minor release of RHEL and allows community contributions (as opposed to just being a rebuild); there's currently CentOS Stream 8 (served from mirror.centos.org via mirrorlist) and CentOS Stream 9 (served via mirror.stream.centos.org via the Fedora mirrormanager infra). So, to recap:
- `install_centos_repos_old` is used for CentOS Linux <= 7
- `install_centos_repos_new` is used for CentOS Linux 8 and CentOS Stream 8
- `install_centos_stream_repos` is used for CentOS Stream 9 (and there is no CentOS Linux 9)

EPEL just tracks the major version of the corresponding CentOS/RHEL release, so e.g. EPEL 8 will work with CentOS Linux 8, CentOS Stream 8 and RHEL 8 (with a tiny caveat that's not relevant here).

Reply:
Current logic looks fine except the is_stream variable doesn't seem used so I'd change parse_epel_release to not return a Tuple and remove the is_stream variable